app/views/exercises/index.html.erb

```diff
@@ -15,7 +15,14 @@
     <% @exercises.each do |exercise| %>
         <tr>
           <td><%= language_image_tag exercise.language %> <%= exercise.language %> </td>
-          <td><%= exercise.title %></td>
+          <td>
+            <% if exercise.solved_by?(current_user) && current_user %>
+              <span class="glyphicon glyphicon-certificate passed-exercise" aria-hidden="true"></span>
```

Comment:
If you invert the condition you grant that solved_by always receives a non null argument.